linux_os/guide/system/accounts/accounts-restrictions/root_logins/no_shelllogin_for_systemaccounts/oval/shared.xml

```diff
@@ -71,7 +71,7 @@
   <!-- Get all /etc/passwd entries having shell defined as OVAL object -->
   <ind:textfilecontent54_object id="object_etc_passwd_entries" version="1">
     <ind:filepath>/etc/passwd</ind:filepath>
-    <ind:pattern operation="pattern match">^(?!root).*:x:([\d]+):[\d]+:[^:]*:[^:]*:(?!\/usr\/sbin\/nologin|\/sbin\/nologin|\/bin\/sync|\/sbin\/shutdown|\/sbin\/halt).*$</ind:pattern>
+    <ind:pattern operation="pattern match">^(?!root).*:x:([\d]+):[\d]+:[^:]*:[^:]*:(?!\/usr\/sbin\/nologin|\/sbin\/nologin|\/bin\/sync|\/sbin\/shutdown|\/sbin\/halt|\/bin\/false|\/usr\/bin\/false).*$</ind:pattern>
```

Comment:
Hi, this change has been reverted previously.
See https://github.com/ComplianceAsCode/content/pull/3767 and its comments.

If this is really needed for Ubuntu, please add a conditional based on the product.

If you agree with the justification for avoiding accepting `/bin/false` and `/usr/bin/false, and willing to go an extra mile. It seems that a warning in the rule would help understand why they are not there.

Reply:
Well, as @cipherboy remind me, that should apply to /bin/sync and, by extension, /sbin/shutdown, /sbin/halt too. And the latters are handled as exceptions in the aforementioned regexp.

Also, the way I see it, the problem of using the account for port forwarding must actually be mitigated by a security rule that locks system accounts. Using this rule for that purpose is a bad fit.

Reply:
There's also an explicit parameter to block port forwarding in general. (`DisableForwarding` being the strongest, but also the `Allow*Forwarding` parameters can be set to `no`).

Reply:
> Well, as @cipherboy reminded me, that should apply to /bin/sync and, by extension, /sbin/shutdown, /sbin/halt too. And the latters are handled as exceptions in the aforementioned regexp.

Yeah, you have a point. And re-reading the article `nologin` is not any better.
 
> Also, the way I see it, the problem of using the account for port forwarding must actually be mitigated by a security rule that locks system accounts. Using this rule for that purpose is a bad fit.

I agree.